File: generator.java
import java.util.*;

public class generator {
    
    public static String generate() {
        Random rand = new Random();
        String str = " ";
        
        for (int i = 0; i < 6; i++) {
            str += (char) (rand.nextInt(26) + 97); // lowercase letter
            str += (char) (rand.nextInt(26) + 65); // uppercase letter
            str += (char) (rand.nextInt(10) + 48); // digit
            str += (char) (rand.nextInt(15) + 33); // special character
        }
        
        List<String> characters = new ArrayList<String>(Arrays.asList(str.split(" ")));
        Collections.shuffle(characters);
        String pass = characters.toString();
        return pass;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("How long do you want the password to be? ");
        int length = sc.nextInt();
        sc.close();
        String result = generate();
        System.out.println(result.substring(0, length));
    }
}

File: generator.py
import random 

def generate():
    string = ""

    for i in range(0,6):
        num = random.randint(97,122)
        string = string + chr(num)
        num = random.randint(65,90)
        string = string + chr(num)
        num = random.randint(48,57)
        string = string + chr(num)
        num = random.randint(33,47)
        string = string + chr(num)

    characters = list(string)
    random_sample = random.sample(characters, len(characters))
    newString = ''.join(random_sample)
    return newString

if __name__ == "__main__":
    length = int(input("How long do you want the password to be? "))
    result = generate()
    print(result[:length])
